### ISSUE TITLE
type.array() throwing validation error.

### ISSUE THREAD

**Issue by xbenjii** [neumino/thinky#221]
Using v2.0.2

**database.js**
``` JavaScript
import Thinky from 'thinky';
import config from '../../config';

var thinky = Thinky(config.database);

export default thinky;
```

**channel.js**
``` JavaScript
import db from '../database';
import User from './user';

var type = db.type;

var Channel = db.createModel('Channel', {
    id: type.number(),
    token: type.string(),
    name: type.string(),
    adminId: type.string(),
    commands: type.array()
});

Channel.hasMany(User, 'admins', 'id', 'adminId');

export default Channel;
```

The actual error

> Document failed validation: The value must be `String`/`Number`/`Boolean`/`Date`
/`Buffer`/`Object`/`Array`/`'virtual'`/`'Point'` for [commands]

If you need any more information, let me know.

**Comment by neumino** [comment]
That's because you need to define the schema of `type.array()`. Something like that should work `commands: type.array().schema(type.string())`.

I'll see if I can have thinky throw a nicer error in this case.

**Comment by xbenjii** [comment]
Must've missed that in the docs, thanks. :)

**Comment by chrisinajar** [comment]
I just ran into this as well, a friendly error would be great. Took me a while to figure out all array's need schemas, even if the schema is "type.any".

I think the schema should default to type.any() to avoid the issue and better match hapijs/joi 's familiar api.

**Comment by neumino** [comment]
@chrisinajar -- Thank you for the feedback. That seems like a reasonable thing to do.

**Comment by neumino** [comment]
Fixed in 2.0.10
